lib/astarte/client/appengine.ex

```diff
@@ -29,24 +29,30 @@ defmodule Astarte.Client.AppEngine do
 
   @jwt_expiry 5 * 60
 
-  @spec new(String.t(), String.t(), String.t()) :: {:ok, t()} | {:error, any}
-  def new(base_api_url, realm_name, private_key) do
-    base_url = Path.join([base_api_url, "appengine", "v1", realm_name])
+  @spec new(String.t(), String.t(), [jwt: String.t()] | [private_key: String.t()]) ::
+          {:ok, t()} | {:error, any}
+  def new(base_api_url, realm_name, private_key: private_key) do
```

Comment:
When something gets passed as a keyword list, that field implicitly becomes an `opts` field (that could possibly be expanded). This usually means that even if right now there could only be a single option, it's not good practice pattern matching on a keyword list (since, differently from e.g. a map, it's order dependent).

What I would expect here is something like:
```elixir
def new(base_api_url, realm_name, private_key: private_key) do
  with {:ok, jwt} <- fetch_jwt(opts) do
  [...]
end

defp fetch_jwt(opts) do
  with :error <- Keyword.fetch(opts, :jwt) do
    generate_jwt(Keyword.get(opts, :private_key)
  end
end

defp generate_jwt(nil = private_key) do
  {:error, :missing_jwt_and_private_key}
end

defp generate_jwt(private_key) do
  credentials = Credentials.appengine_all_access_credentials(expiry: @jwt_expiry)
  [...]
end
```